app/controllers/admin/secrets_controller.rb

```diff
@@ -66,10 +82,26 @@ def find_secret
     @secret = SecretStorage.read(key)
   end
 
+  def generate_deploy_group_list
+    @deployment_group_list = []
+    DeployGroup.all.map do |group|
```

Comment:
`@deployment_group_list =  DeployGroup.all.map { |g| [g.environment.permalink, g.permalink] }`

only used in the view ... maybe keep it there for simplicity ... not necessary for index/destroy/update